### PR TITLE
fix: return versionId when delete non-exist ver [S3C-368]

### PR DESCRIPTION
**Pull request by electrachong**
- tests to confirm we do not create delete markers for deleting specific versions
- make sure version ID is sent back for requests trying to delete non-existing version

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine

File: lib/api/objectDelete.js [hunk 1]
```diff
@@ -125,12 +122,19 @@ export default function objectDelete(authInfo, request, log, cb) {
...
             }
-            log.debug('error processing request', { error: err,
-                method: 'objectDelete' });
-            return cb(err, resHeaders);
+            log.trace('received request to delete non-existing version');
+            if (reqVersionId) {
+                resHeaders['x-amz-version-id'] = reqVersionId === 'null' ?
+                    reqVersionId : versionIdUtils.encode(reqVersionId);
+            }
+            pushMetric('deleteObject', log, { authInfo, bucket: bucketName,
```
Comment (bennettbuchanan) [inline lib/api/objectDelete.js]: Have we validated this is the correct behavior? In any case, we should probably add an Integration test for this case.

**Comment by electrachong** [comment]
end-to-end passed

File: lib/api/objectDelete.js (hunk 1 above)
Comment (electrachong) [inline lib/api/objectDelete.js]: I think I will have to check with @rahulreddy whether we should increment the counter for the  'deleteObject' action, even though we don't delete any actual object/version in this scenario.

File: tests/functional/aws-node-sdk/test/versioning/multiObjectDelete.js [hunk 2]
```diff
@@ -136,6 +142,35 @@ describe('Multi-Object Versioning Delete Success', function success() {
...
+                ({ Key: obj.Key, VersionId: obj.VersionId }));
+            const prevVersion = objects[0].VersionId;
+            objects[0].VersionId = nonExistingId;
+            return s3.deleteObjectsAsync({
+                Bucket: bucketName,
+                Delete: {
+                    Objects: objects,
+                },
+            }).then(res =>
+                s3.deleteObjectAsync({
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/multiObjectDelete.js]: why are you doing the clean up delete in here?

File: lib/api/objectDelete.js (hunk 1 above)
Comment (electrachong) [inline lib/api/objectDelete.js]: I checked with Rahul and he said we don't increment the counter for the api if the error is NoSuchKey or NoSuchVersion. I removed the pushMetric call here.

File: tests/functional/aws-node-sdk/test/versioning/multiObjectDelete.js (hunk 2 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/multiObjectDelete.js]: I moved the clean-up to the after each callback.

**Comment by electrachong** [comment]
@ironman-machine

**Comment by electrachong** [comment]
end-to-end passed
http://ci.ironmann.io/gh/scality/Integration/13008

**Pull request merged**